Podcast/MiniPlayerView.swift

```diff
@@ -0,0 +1,90 @@
+
+import UIKit
+
+protocol MiniPlayerViewDelegate: class {
+    func miniPlayerViewDidTapPlayPauseButton()
+    func miniPlayerViewDidTapExpandButton()
+}
+
+class MiniPlayerView: UIView {
+    
+    let miniPlayerHeight: CGFloat = 58
+    let MarginSpacing: CGFloat = 24
+    let ButtonDimension: CGFloat = 24
+    let arrowYValue: CGFloat = 25
+    let arrowWidth: CGFloat = 16
+    let arrowHeight: CGFloat = 8
+    let labelYVal: CGFloat = 12
+    let labelHeight: CGFloat = 34
+    
+    var arrowButton: UIButton!
+    var playPauseButton: UIButton!
+    var episodeTitleLabel: UILabel!
+    
+    var transparentMiniPlayerEnabled: Bool = true
+    
+    weak var delegate: MiniPlayerViewDelegate?
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        
+        self.frame.size.height = miniPlayerHeight
+        backgroundColor = UIColor.colorFromCode(0xf0f1f4)
+        
+        if !UIAccessibilityIsReduceTransparencyEnabled() && transparentMiniPlayerEnabled {
+            
+            self.backgroundColor = UIColor.clear
+            
+            let blurEffect = UIBlurEffect(style: UIBlurEffectStyle.extraLight)
+            let blurEffectView = UIVisualEffectView(effect: blurEffect)
+            
+            blurEffectView.frame = self.bounds
+            blurEffectView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
+            
+            self.addSubview(blurEffectView)
+        }
+        
+        arrowButton = UIButton(frame: CGRect(x: MarginSpacing, y: arrowYValue, width: arrowWidth, height: arrowHeight))
+        arrowButton.addTarget(self, action: #selector(arrowButtonTapped), for: .touchUpInside)
+        arrowButton.setBackgroundImage(#imageLiteral(resourceName: "down_arrow"), for: .normal)
+        addSubview(arrowButton)
+        
+        playPauseButton = UIButton(frame: CGRect(x: frame.size.width - MarginSpacing - ButtonDimension, y: 17, width: ButtonDimension, height: ButtonDimension))
+        playPauseButton.adjustsImageWhenHighlighted = false
+        playPauseButton.addTarget(self, action: #selector(playPauseButtonTapped), for: .touchUpInside)
+        addSubview(playPauseButton)
+        
+        episodeTitleLabel = UILabel(frame: CGRect(x: arrowButton.frame.maxX + MarginSpacing, y: labelYVal, width: playPauseButton.frame.minX - 2 * MarginSpacing - arrowButton.frame.maxX, height: 34))
+        episodeTitleLabel.numberOfLines = 2
+        episodeTitleLabel.textAlignment = .left
+        episodeTitleLabel.lineBreakMode = .byWordWrapping
+        addSubview(episodeTitleLabel)
+    }
+    
+    func updateUI() {
+        if Player.sharedInstance.isPlaying {
```

Comment:
UI should be updated by the episode vc that passes in the required parameters to update the UI